src/librustc/ty/mod.rs

```diff
@@ -3085,16 +3085,31 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     /// its supposed definition name (`def_name`). The method also needs `DefId` of the supposed
     /// definition's parent/scope to perform comparison.
     pub fn hygienic_eq(self, use_name: Ident, def_name: Ident, def_parent_def_id: DefId) -> bool {
-        self.adjust_ident(use_name, def_parent_def_id, hir::DUMMY_HIR_ID).0 == def_name.modern()
+        // We could use `Ident::eq` here, but we deliberately don't. The name
+        // comparison fails frequently, and we want to avoid the expensive
+        // `modern()` calls required for the span comparison whenever possible.
+        use_name.name == def_name.name &&
+        self.adjust_ident(use_name, def_parent_def_id).span.ctxt() == def_name.modern().span.ctxt()
     }
 
-    pub fn adjust_ident(self, mut ident: Ident, scope: DefId, block: hir::HirId) -> (Ident, DefId) {
-        ident = ident.modern();
-        let target_expansion = match scope.krate {
+    #[inline]
+    fn target_expansion(self, scope: DefId) -> Mark {
```

Comment:
```suggestion
    fn expansion_that_defined(self, scope: DefId) -> Mark {
```
(`let target_expansion` wasn't an example of good variable naming.)

Also, inline is usually unnecessary on private functions?